test/distributed/test_pg_wrapper.py

```diff
@@ -212,177 +212,175 @@ def _test_collective_shape_mismatch(self, wrapper_pg, use_cuda=False):
-        if with_new_group:
-            pg = c10d.new_group(backend="gloo")
-        else:
-            _pg = c10d.ProcessGroupGloo(
-                store, self.rank, self.world_size, self.opts(timeout=timeout)
+    class ProcessGroupNCCLWrapperTest(AbstractProcessGroupWrapperTest):
+        def setUp(self):
+            super(AbstractProcessGroupWrapperTest, self).setUp()
+            self._spawn_processes()
+            # NCCL_BLOCKING_WAIT overrides NCCL_ASYNC_ERROR_HANDLING hence tests
+            # that use NCCL_BLOCKING_WAIT will test it as expected.
+            os.environ["NCCL_ASYNC_ERROR_HANDLING"] = "1"
+
+        @property
+        def world_size(self) -> int:
+            return 2
+
+        def _create_wrapper_pg(self, with_new_group=False, timeout=10.0):
+            store = c10d.FileStore(self.file_name, self.world_size)
+            c10d.init_process_group(
+                backend="nccl",
+                rank=self.rank,
+                world_size=self.world_size,
+                store=store,
+                timeout=timedelta(seconds=timeout),
             )
-            pg = c10d._create_process_group_wrapper(
-                _pg,
-                "unused",
-                store,
-                self.rank,
-                self.world_size,
-                timeout=timeout,
+            if with_new_group:
+                pg = c10d.new_group(backend="nccl", timeout=timedelta(seconds=timeout))
+            else:
+                _pg = c10d.ProcessGroupNCCL(
+                    store, self.rank, self.world_size, timeout=timedelta(seconds=timeout)
+                )
+                pg = c10d._create_process_group_wrapper(
+                    _pg,
+                    "unused",
+                    store,
+                    self.rank,
+                    self.world_size,
+                    timeout=timeout,
+                )
+            return pg
+
+        @requires_nccl()
+        @skip_if_lt_x_gpu(2)
+        @sandcastle_skip_if(torch.cuda.device_count() < 2, "NCCL test requires 2+ GPUs")
+        def test_collective_hang(self):
+            pg = self._create_wrapper_pg(timeout=2.0)
+            self._test_collective_hang(pg)
+
+        # NOTE: these tests are separated by debug level instead of combined into
+        # one due to https://github.com/pytorch/pytorch/issues/55967, they can be
+        # combined after that is resolved.
+        @requires_nccl()
+        @skip_if_lt_x_gpu(2)
+        @with_dist_debug_levels(levels=["DETAIL"])
+        @sandcastle_skip_if(torch.cuda.device_count() < 2, "NCCL test requires 2+ GPUs")
+        def test_collectives_op_mismatch_debug_mode(self):
+            pg = self._create_wrapper_pg(with_new_group=True)
+            self._test_collectives_op_mismatch(pg, use_cuda=True)
+
+        @requires_nccl()
+        @skip_if_lt_x_gpu(2)
+        @with_dist_debug_levels(levels=["OFF"])
+        @sandcastle_skip_if(torch.cuda.device_count() < 2, "NCCL test requires 2+ GPUs")
```

Comment:
Not sure I understand why we need both `sandcastle_skip_if` and then skip_if_lt_x_gpu_2 as well here? Shouldn't we only need the sandcastle one?